pkg/controller/replication_controller.go

```diff
@@ -327,6 +327,7 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	obj, exists, err := rm.controllerStore.Store.GetByKey(key)
 	if !exists {
 		glog.Infof("Replication Controller has been deleted %v", key)
+		rm.expectations.DeleteExpectations(key)
```

Comment:
hm. That gets complicated with expectations. Maybe this is better behavior. Ideally I'd like it to make one last attempt to sync; so if you set replicas=0 and then delete the rc, it would in fact delete all the replicas. But if you set replicas=5, delete the RC & add it again with a new name and replicas=17, RC Manager should not keep the old one around to fight with the new one...